ttnn/cpp/ttnn/operations/transformer/concatenate_heads/concatenate_heads.cpp

```diff
@@ -31,17 +31,16 @@ struct ConcatenateHeads : public ttnn::operations::experimental::transformer::NL
     }
 
     std::vector<tt::tt_metal::LegacyShape> compute_output_shapes(const std::vector<Tensor>& input_tensors) const {
-        std::vector<tt::tt_metal::LegacyShape> output_shape_vec;
         const auto& input_tensor = input_tensors.at(0);
-        const ttnn::types::Shape input_shape = input_tensor.get_shape();
-        const ttnn::types::Shape padded_input_shape = input_shape.with_tile_padding();
-
-        auto batch_size = input_shape[0];
-        auto num_heads = input_shape[1];
-        auto sequence_size = input_shape[2];
-        auto padded_sequence_size = padded_input_shape[2];
-        auto head_size = input_shape[3];
-        auto padded_head_size = padded_input_shape[3];
+        const ttnn::SimpleShape input_logical_shape = input_tensor.get_logical_shape();
+        const ttnn::SimpleShape input_padded_shape = input_tensor.get_padded_shape();
+
+        auto batch_size = input_logical_shape[0];
```

Comment:
You're not changing the logic here, but there is no bounds check on this shape indexing. Since we're opening this file anyways, what are your thoughts on validating the `input_shape` rank in the validate function?

Reply:
Added